### PR TITLE
#1895 Publish v12  Docs at different URL 

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1895
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by srikant-ch5** [comment]
Hi @matthoward366 to have separate documentation for v12 and v13 I am using a third party python utility [mike](https://pypi.org/project/mike/). After deploying it will show two version dropdown like below  
Screenshot 2024-05-08 at 1 31 29 PM

https://srikant-ch5.github.io/canvas/v13/

**Review by tomlyn — CHANGES_REQUESTED** [review]
Why have you renamed the 'pages' folder to be 'docs' - it doesn't make any sense to have a `docs/docs` folder structure.

**Comment by srikant-ch5** [comment]
> Why have you renamed the 'pages' folder to be 'docs'? it doesn't make any sense to have a `docs/docs` folder structure.

Hi @tomlyn initially I tried with 'pages' folder name but mike commands were not able to make changes to gh-pages branch, after I renamed it to docs it was able to execute.

**Comment by tomlyn** [comment]
@srikant On Slack you said
> Hi Craig regarding [v12 and v13 Docs PR](https://github.com/elyra-ai/canvas/pull/1931#pullrequestreview-2046295966)  as per mike the folder name should be docs otherwise its not deploying pages in separate folder to tag under appropriate version.
> dropdown will look like this https://srikant-ch5.github.io/canvas/v13/ . Please let me know whether I should change foldername from docs/docs to [some other name]/docs

If you are 100% sure mike cannot be configured with a different directory structure then can we use `mkdocs/docs` instead?

**Comment by srikant-ch5** [comment]
> @srikant On Slack you said
> 
> > Hi Craig regarding [v12 and v13 Docs PR](https://github.com/elyra-ai/canvas/pull/1931#pullrequestreview-2046295966)  as per mike the folder name should be docs otherwise its not deploying pages in separate folder to tag under appropriate version.
> > dropdown will look like this https://srikant-ch5.github.io/canvas/v13/ . Please let me know whether I should change foldername from docs/docs to [some other name]/docs
> 
> If you are 100% sure mike cannot be configured with a different directory structure then can we use `mkdocs/docs` instead?

Hi @tomlyn Could you please review this PR  I have included mike and changes from v13 ?

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -42,7 +52,7 @@ extra_javascript:
 edit_uri: edit/main/docs/
 
 # Documentation and theme
-copyright: 'Copyright  &copy; 2024 Elyra Authors, an open source project and all are welcome to contribute! Maintained by the Elyra Authors Team'
+copyright: 'Copyright  &copy; 2024 Elyra Authors, an open source project! Maintained by the Elyra Authors Team'
```
Comment (tomlyn) [inline docs/mkdocs.yml:55]: This is also a change I made. This shouldn't show as your change.

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -17,19 +17,29 @@ site_name: Elyra Canvas
...
       - navigation.tabs
       - navigation.indexes
 
+extra:
+  version:
+    provider: mike
+  social:
+    - icon: fontawesome/brands/github
+      link: http://github.com/elyra-ai/canvas
+      name: Elyra Canvas on GitHub
```
Comment (tomlyn) [inline docs/mkdocs.yml:41]: The changes above are also edits I made. This shouldn't show as your change.

**Review by tomlyn — COMMENTED** [review]
Can you update the README with any new instructions on what commands need to be run just like you added instructions when `mkdocs` was introduced?

File: docs/mkdocs.yml [hunk 3]
```diff
@@ -17,19 +17,29 @@ site_name: Elyra Canvas
...
 
+# Configuration
 theme:
     name: material
     highlightjs: true
     logo: assets/FlowIcon.png
     favicon: assets/FlowIcon.png
     palette:
         scheme: default
+        primary: black
```
[the remark this answers is not in the document]
Comment (srikant-ch5) [reply docs/mkdocs.yml:28]: This change is in 12.x branch as in this mkdocs.yml above change was not present so to keep it consistent with v13 I have updated this, hope this is fine.

File: docs/mkdocs.yml (hunk 1 above)
Comment (srikant-ch5) [reply docs/mkdocs.yml:55]: This change is in 12.x branch as in this mkdocs.yml above change was not present so to keep it consistent with v13 I have updated this, hope this is fine.

File: docs/mkdocs.yml (hunk 2 above)
Comment (srikant-ch5) [reply docs/mkdocs.yml:41]: same as above comment.

**Comment by srikant-ch5** [comment]
> Can you update the README with any new instructions on what commands need to be run just like you added instructions when `mkdocs` was introduced?



> Can you update the README with any new instructions on what commands need to be run just like you added instructions when `mkdocs` was introduced?

Sure. I will include it.

**Comment by tomlyn** [comment]
Hi @srikant-ch5 

As agreed with @matthoward366, in the scrum today we don't need to make this change for v12.  The important thing is that we have multi-version support for `mkdocs` for future releases (that is for v13.0.0 and beyond).

**Comment by srikant-ch5** [comment]
Hi @tomlyn Thanks for the update. Closing this PR.

**Pull request closed without merging**